src/type-list-of.c

```diff
@@ -3,7 +3,7 @@
 
 // [[ include("utils.h") ]]
 SEXP new_list_of(SEXP x, SEXP ptype) {
-  if (TYPEOF(x) != VECSXP) {
+  if (!vec_is_list(x)) {
```

Comment:
This explicit support for S3 lists makes me wonder if we should zap the attributes before initialising the list-of.

Reply:
We probably should.

Reply:
The R level one does, so yea probably. I'll update. 

```
attributes(vctrs::new_list_of(structure(list(), x = 1)))
#> $ptype
#> logical(0)
#> 
#> $class
#> [1] "vctrs_list_of" "vctrs_vctr"
```